proto/telemetry.proto

```diff
@@ -111,3 +111,41 @@ message FrontendReport {
 message CompactorReport {
   ReportBase base = 1;
 }
+
+enum TelemetryEventStage {
+  Telemetry_Event_Stage_UNSPECIFIED = 0;
+  Telemetry_Event_Stage_CreateStreamJob = 1;
+  Telemetry_Event_Stage_UpdateStreamJob = 2;
+  Telemetry_Event_Stage_DropStreamJob = 3;
+  Telemetry_Event_Stage_Query = 4;
+  Telemetry_Event_Stage_Recovery = 5;
+}
+
+enum TelemetryConnectorDirection {
+  Telemetry_Connector_Direction_UNSPECIFIED = 0;
+  Telemetry_Connector_Direction_Source = 1;
+  Telemetry_Connector_Direction_Sink = 2;
+}
+
+message EventMessage {
+  // tracking_id is persistent in meta data
+  string tracking_id = 1;
+  // session_id is reset every time node restarts
+  string session_id = 2;
+  // event_time is when the event is created
+  uint64 event_time = 3;
+  // event_stage describes in which process the event happens
+  TelemetryEventStage event_stage = 4;
+  // feature_name is the name of the feature triggered the event
+  string feature_name = 5;
+  // connector_name is the name of the connector involves
+  optional string connector_name = 6;
+  // connector_direction is the direction of data flow, can be source or sink
+  optional TelemetryConnectorDirection connector_direction = 10;
```

Comment:
I think it's better to record the kind of **database object** here, that is - table, source, sink, or MV, or even more such as schema, user, connection, .etc.

Reply:
make sense, let me fix it

Reply:
I noticed you created a new `enum TelemetryDatabaseComponents`. However, "database **object**" is a term and it refers to an entity of specific semantics. [Google it](https://www.google.com/search?q=database+objects) for more references.